packages/vite/src/node/plugins/resolve.ts

```diff
@@ -310,7 +309,13 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
           asSrc &&
           depsOptimizer &&
           !options.scan &&
-          (res = await tryOptimizedResolve(depsOptimizer, id, importer))
+          (res = await tryOptimizedResolve(
```

Comment:
I am not very sure this `tryOptimizedResolve` call would trigger some edge cases that return unexpected result 🤔.